index.bs

```diff
@@ -7652,6 +7652,122 @@ To <dfn abstract-op>Create a new supplemental public key record</dfn>, perform t
+
+    <dl dfn-type="dict-member" dfn-for="CollectedClientConfirmationData">
+        :  <dfn>confirmationPrompt</dfn> member
+        :: The full context of the confirmation to give.
+    </dl>
+
+
+    <xmp class="idl">
+        dictionary CollectedClientAdditionalConfirmationData {
+            required USVString confirmationPrompt;
+        };
+    </xmp>
+
+    The {{CollectedClientAdditionalConfirmationData}} dictionary contains the following
+    fields:
+
+    <dl dfn-type="dict-member" dfn-for="CollectedClientAdditionalConfirmationData">
+        :  <dfn>confirmationPrompt</dfn> member
+        :: The full context of the confirmation to give.
+    </dl>
+
+: Client extension output
+:: Returns the authenticator extension output string UTF-8 decoded into a USVString.
+    <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientOutputs {
+      USVString confirmationPrompt;
+    };
+    </xmp>
+
+: Authenticator extension input
+:: The client provides the extension input encoded as a CBOR text string (major type 3).
+
+<pre>
+CDDL:
+confirmationPromptInput = (tstr)
+</pre>
+
+: Authenticator extension processing
+:: The authenticator supporting this extension MUST display the confirmation prompt to the user
+    before performing either [=user verification=] or [=test of user
+    presence=]. The authenticator MAY wrap lines that are too wide to be shown if needed.
+
+: Authenticator extension output
+:: If the authenticator supports this extension (and only then): A single CBOR string, representing the
+    confirmation prompt.
+
+<pre>
+CDDL:
+confirmationPromptOutput = (tstr)
+</pre>
+
+#### `confirmation` Extension Output Verification Procedures #### {#sctn-confirmation-extension-verification}
+
+Verifying the <code>[=confirmation=]</code> extension output is performed by the [=[RP]=] whenever a the use of <code>[=confirmatione=]</code> was requested.
+
+##### Authentication (`get()`) ##### {#sctn-confirmation-extension-verification-get}
+
+This extension is only specified to be used in conjunction with a {{CredentialsContainer/get()|navigator.credentials.get()}} call.
+
+If the [=[RP]=] requested the `confirmation` extension in a {{CredentialsContainer/get()|navigator.credentials.get()}} call, then the below verification step are performed in the context of [step 19](#authn-ceremony-verify-extension-outputs)
+of [[#sctn-verifying-assertion]] using these variables established therein: |credential|, |clientExtensionResults|, |authData|, |hash|, and |credentialRecord|.
+[=[RP]=] policy may specify whether a response without a `confirmation` extension output is acceptable.
+
+1. Verify that the {{AuthenticationExtensionsClientOutputs/confirmation}} member of |clientExtensionResults| exists.
+1. Verify that the {{AuthenticationExtensionsClientOutputs/confirmation}} member contains the confirmation prompt that is expected (i.e., that was used when requesting the `confirmation` extension.
```

Comment:
As noted earlier, this should rather be using the authenticator extension output or the `clientDataJSON` entry, right? Maybe look for the authenticator extension output first, and fall back to the `clientDataJSON` entry if the authenticator extension output is absent?

```suggestion
[=[RP]=] policy may specify whether a response without a `confirmation` extension output is acceptable.

1. Verify that the `confirmation` key exists in the [=authData/extensions=] in |authData|.
1. Verify that the `confirmation` value in the [=authData/extensions=] in |authData| equals the confirmation prompt that is expected (i.e., that was used when requesting the `confirmation` extension).
1. (TODO: Fall back to <code>|credential|.{{PublicKeyCredential/response}}.{{AuthenticatorResponse/clientDataJSON}}</code> entry if authenticator extension output is absent?)
```

Reply:
Done.